src/client/runner/automation/get-element.js

```diff
@@ -0,0 +1,73 @@
+import hammerhead from '../deps/hammerhead';
+import testCafeCore from '../deps/testcafe-core';
+import cursor from './cursor';
+
+var browserUtils  = hammerhead.utils.browser;
+var positionUtils = testCafeCore.positionUtils;
+var domUtils      = testCafeCore.domUtils;
+
+
+function getElementFromPoint (x, y) {
+    var cursorVisible = cursor.visible;
+
+    if (cursorVisible)
+        cursor.hide();
+
+    var topElement = positionUtils.getElementFromPoint(x, y);
+
+    if (cursorVisible)
+        cursor.show();
+
+    return topElement;
+}
+
+function ensureImageMap (imgElement, areaElement) {
+    var mapElement = domUtils.closest(areaElement, 'map');
+
+    return mapElement && mapElement.name === imgElement.useMap.substring(1) ? areaElement : imgElement;
+}
+
+function findElementOrNonEmptyChildFromPoint (x, y, element) {
+    var topElement = positionUtils.getElementFromPoint(x, y);
+
+    if (topElement && topElement === element || domUtils.containsElement(element, topElement) && topElement.textContent.length)
+        return topElement;
+
+    return null;
+}
+
+export function fromPoint (x, y, expectedElement) {
+    var topElement = getElementFromPoint(x, y);
+
+    if (!expectedElement || !domUtils.isDomElement(expectedElement) || topElement === expectedElement || !topElement)
+        return topElement;
+
+    // NOTE: T299665 - Incorrect click on image with associated map element in Mozilla
+    // All the browsers return <area> element by document.getElementFromPoint, but
+    // firefox returns <img> element. We should implement this for firefox too.
+    var isImageMapArea = expectedElement.tagName.toLowerCase() === 'area' && topElement.tagName.toLowerCase() === 'img';
+
+    if (browserUtils.isFirefox && isImageMapArea)
+        return ensureImageMap(topElement, expectedElement);
+
+
+    //NOTE: try to find multiline link by her rectangle (T163678)
+    var isLinkOrChildExpected        = expectedElement.tagName.toLowerCase() === 'a' || domUtils.getParents(expectedElement, 'a').length;
+    var isTopElementChildOfLink      = isLinkOrChildExpected && domUtils.containsElement(expectedElement, topElement) && topElement.textContent.length;
+    var shouldSearchForMultilineLink = isLinkOrChildExpected && !isTopElementChildOfLink && expectedElement.textContent.length;
+
```

Comment:
Make these long assignments multiline, splitting them on operators, e.g.:
```js
    var isLinkOrChildExpected  = expectedElement.tagName.toLowerCase() === 'a' ||   
                                 domUtils.getParents(expectedElement, 'a').length;

      var isTopElementChildOfLink = isLinkOrChildExpected &&  
                                    domUtils.containsElement(expectedElement, topElement) && 
                                    topElement.textContent.length;
```
Note that multiline declarations should be split with the blank line